src/libraries/System.Private.CoreLib/src/System/Threading/Interlocked.cs

```diff
@@ -222,6 +226,65 @@ public static UIntPtr Exchange(ref UIntPtr location1, UIntPtr value)
             return (UIntPtr)Exchange(ref Unsafe.As<UIntPtr, int>(ref location1), (int)value);
 #endif
         }
+
+        /// <summary>Sets a variable of the specified type <typeparamref name="T"/> to a specified value and returns the original value, as an atomic operation.</summary>
+        /// <param name="location1">The variable to set to the specified value.</param>
+        /// <param name="value">The value to which the <paramref name="location1"/> parameter is set.</param>
+        /// <returns>The original value of <paramref name="location1"/>.</returns>
+        /// <exception cref="NullReferenceException">The address of location1 is a null pointer.</exception>
+        /// <exception cref="NotSupportedException">An unsupported <typeparamref name="T"/> is specified.</exception>
+        /// <typeparam name="T">
+        /// The type to be used for <paramref name="location1"/> and <paramref name="value"/>.
+        /// This type must be a reference type, an enum type (i.e. typeof(T).IsEnum is true), or a primitive type (i.e. typeof(T).IsPrimitive is true).
+        /// </typeparam>
+        [return: NotNullIfNotNull(nameof(location1))]
```

Comment:
```suggestion
        [Intrinsic]
        [return: NotNullIfNotNull(nameof(location1))]
```
This should still get the JIT handling to reduce diffs.